src/app/backend/handler/apihandler.go

```diff
@@ -220,22 +243,22 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 
 	apiV1Ws.Route(
 		apiV1Ws.POST("/appdeployment").
-			To(apiHandler.handleDeploy).
+			To(apiHandler.injectApiClient(apiHandler.handleDeploy)).
```

Comment:
This style of invocation concerns me... it means the request starts out as the shared in-cluster-config user unless overridden by a call to injectApiClient. That means if a new API call is added, and the call to injectApiClient is forgotten, an unintentional escalation can occur.

If the goal is to make the dashboard safe to expose to multiple users without risking permission escalations, the override should go in the other direction... by default, make a request with no credentials, add in user-provided credentials if present, and only use the in-cluster-config if a particular use of apiHandler explicitly chooses to use the dashboard's credentials.